### DISCUSSION TITLE
SEBE simulating diffused radiation

### DISCUSSION THREAD

**Discussion by klyu6517** [UMEP-dev/UMEP#594] [General]
Dear developers,

Thank you for this wonderful toolbox! I am currently trying to simulate daylight/sunlight availability on building facades accounting the potential influence of urban morphology. I came across SEBE within UMEP, and it seems very relevant. I was wondering that apart from calculating direct solar energy on building roof and walls, is it possible to also calculate the diffused radiation from the sky? This would be of interests in researchers in the lighting field. If so, how can I do it? Thank you very much for your answer in advance.

Best,
Kun

**Comment by biglimp** [comment]
At the moment this is not included as an output, event though it is calculated. I willl try to guide you through some steps to output this information by some so-called hard coding. The instructions below is how to include output for xy areas, i.e. roofs and ground.

1. First you need to locate where the plugin is installed on your computer. This depends on OS. This guide is for Windows. On my PC, I find the plugins in `C:\Users\my_user_name\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins`. The SEBE tool is located in `processing_umep\processor`. The you find a file called **sebe_algorithm.py**. That is the main file for the tool.
2. Open this file in a text editor, preferably one that show you line numbers e.g. Notepad++. Go to line 346. That is the function that do all the calculations. As you see there is a variable called seberesult that comes out. This is a so called dictionary that we can use to store our new diffuse grid. Also note on line 360 a function called saveraster. We can use this to save our new raster.
3. Add on line 355 `saveraster(self.gdal_dsm, outputDir + '/Diffuseyearroof.tif', seberesult["diffusegrid"])`. Make sure that you put is the same indention as line 354.
4. Now we need to create the diffuse grid which will happen in `processing_umep\functions\SEBEfiles\SEBE_2025a_calc_forprocessing.py`. Open this file in a text editor.
5. Fist we need to create an empty grid. In line 15, add `diffusegrid = np.copy(Knight)`
6. Next, add info to the new grid on line 107: `diffusegrid = np.copy(diffusegrid + D)`
7. Next on line 148, add: `diffusegrid /= 1000`
8. Finally adjust line 153: `seberesult = {'Energyyearroof': Energyyearroof, 'Energyyearwall': Energyyearwall, 'vegdata': vegdata, 'diffusegrid': diffusegrid}`

That should be it. Save and close your files, restart QGIS, rerun SEBE and there should be a new output in the outputfolder specified.

Good luck! /F

**Comment by klyu6517** [comment]
Thank you very much for your reply, Fredrik! I was wondering if it is possible to calculate the diffused radiation on building walls? Similar like the 'Energyyearwall.txt' output file but for diffused radiation.

**Reply by biglimp** [reply]
a bit more complicated. Try this first...
